### PR TITLE
completed task 10

### PR DESCRIPTION
**Pull request by Pumpkin92**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -311,3 +311,30 @@ describe("/api/comments/:comment_id", () => {
...
   });
 });
+
+describe("/api/users", () => {
+  test("GET 200 Responds with an array of objects containing all users", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        body.forEach((user) => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:321]: don't forget you need a length check here to prevent a false positive on this test where the array is empty

File: __tests__/app.test.js [hunk 2]
```diff
@@ -311,3 +311,30 @@ describe("/api/comments/:comment_id", () => {
...
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+  test("GET 404 Responds with a 404 error and not found when URL does not match an endpoint", () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:332]: this technically isn't necessary as its the same as your previous tests that check any invalid route

File: controllers/users-controller.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const { fetchUsers } = require("../models/users.model");
+
+function getUsers(req, res, next) {
+  fetchUsers()
+    .then((users) => {
+      res.status(200).send(users);
```
Comment (roisinoreilly) [inline controllers/users-controller.js:6]: don't forget to send data back on a meaningful key please!

**Review by roisinoreilly — COMMENTED** [review]
nice work on this ticket, I've just left some comments for you to have a look at and think about :)
